Content.Shared/Pinpointer/SharedPinpointerSystem.cs

```diff
@@ -87,6 +130,36 @@ public void SetDistance(EntityUid uid, Distance distance, PinpointerComponent? p
         Dirty(uid, pinpointer);
     }
 
+    /// <summary>
+    /// Stores the located target on the pinpointer if it's not already stored.
+    /// Adds the Trackable component to the target so it can keep track of all the pinpointers tracking it.
+    /// </summary>
+    public void StoreTarget(EntityUid? target, EntityUid pinpointer, PinpointerComponent component, EntityUid user)
+    {
+        if (target == null)
+        {
+            _popup.PopupEntity(Loc.GetString("targeting-pinpointer-failed"), user, user);
+            return;
+        }
+
+        if (component.StoredTargets.Contains(target.Value))
+            return;
+
+        component.StoredTargets.Add(target.Value);
+        EnsureComp<TrackableComponent>(target.Value, out var trackable);
+        trackable.TrackedBy.Add(pinpointer);
+        Dirty(pinpointer, component);
+    }
+
+    /// <summary>
+    ///     Removes a target from the target list.
+    /// </summary>
+    public void RemoveTarget(EntityUid uid, PinpointerComponent component, EntityUid tracker)
+    {
+        component.StoredTargets.Remove(uid);
+        Dirty(tracker, component);
+    }
+
```

Comment:
You never validate if it's removed from the collection before dirtying (though I don't think you should be networking it anyway). Same with using a set instead for adding if you want to treat it as a set.